src/jdk.management/share/native/libmanagement_ext/DiagnosticCommandImpl.c

```diff
@@ -173,8 +177,14 @@ Java_com_sun_management_internal_DiagnosticCommandImpl_getDiagnosticCommandInfo
   }
   jmm_interface->GetDiagnosticCommandInfo(env, commands, dcmd_info_array);
   for (i=0; i<num_commands; i++) {
+      // Ensure capacity for 6 + 3 local refs:
+      //  6 => jname, jdesc, jimpact, cmd, args, obj
+      //  3 => permission class, name, action
+      (*env)->PushLocalFrame(env, 6 + 3);
+
+      cmd = (*env)->GetObjectArrayElement(env, commands, i);
       args = getDiagnosticCommandArgumentInfoArray(env,
-                                                   (*env)->GetObjectArrayElement(env,commands,i),
+                                                   cmd,
                                                    dcmd_info_array[i].num_arguments);
       if (args == NULL) {
```

Comment:
Aren't you missing the PopLocalFrame for this return path?

Reply:
Thanks, David. Added a call to `PopLocalFrame` here and in various other places where we have a short return.